main.go

```diff
@@ -710,10 +710,15 @@ func contextToConfig(c *cli.Context) *core.Config {
 		opts = append(opts, core.WithLogLevel(log.LogInfo))
 	}
 
-	listen := c.String("listen")
-	if listen != "" {
-		opts = append(opts, core.WithListenAddress(listen))
+	publicListen := c.String("public-listen")
+	if publicListen != "" {
```

Comment:
We should use the API for that.
```
if c.IsSet(publicListen.Name) {
   opts = append( opts, core.WithPrivateListenAddress(c.String(publicListen.Name)))
```
where `publicListen` is the name of the flag.